worlds/zork_grand_inquisitor/world.py

```diff
@@ -0,0 +1,210 @@
+from typing import Any, Dict, List, Set, Tuple
+
+from BaseClasses import Item, ItemClassification, Location, Region, Tutorial
+
+from worlds.AutoWorld import WebWorld, World
+
+from .data.item_data import item_data, ZorkGrandInquisitorItemData
+from .data.location_data import location_data, ZorkGrandInquisitorLocationData
+from .data.region_data import region_data
+
+from .data_funcs import (
+    item_names_to_id,
+    item_names_to_item,
+    location_names_to_id,
+    item_groups,
+    items_with_tag,
+    location_groups,
+    locations_by_region,
+    location_access_rule_for,
+    entrance_access_rule_for,
+)
+
+from .enums import (
+    ZorkGrandInquisitorEvents,
+    ZorkGrandInquisitorItems,
+    ZorkGrandInquisitorLocations,
+    ZorkGrandInquisitorRegions,
+    ZorkGrandInquisitorTags,
+)
+
+from .options import ZorkGrandInquisitorOptions
+
+
+class ZorkGrandInquisitorItem(Item):
+    game = "Zork Grand Inquisitor"
+
+
+class ZorkGrandInquisitorLocation(Location):
+    game = "Zork Grand Inquisitor"
+
+
+class ZorkGrandInquisitorWebWorld(WebWorld):
+    theme: str = "stone"
+
+    tutorials: List[Tutorial] = [
+        Tutorial(
+            "Multiworld Setup Guide",
+            "A guide to setting up the Zork Grand Inquisitor randomizer connected to an Archipelago Multiworld",
+            "English",
+            "setup_en.md",
+            "setup/en",
+            ["Serpent.AI"],
+        )
+    ]
+
+
+class ZorkGrandInquisitorWorld(World):
+    """
+    Zork: Grand Inquisitor is a 1997 point-and-click adventure game for PC.
+    Magic has been banned from the great Underground Empire of Zork. By edict of the Grand Inquisitor Mir Yannick, the
+    Empire has been sealed off and the practice of mystic arts declared punishable by "Totemization" (a very bad thing).
+    The only way to restore magic to the kingdom is to find three hidden artifacts: The Coconut of Quendor, The Cube of
+    Foundation, and The Skull of Yoruk.
+    """
+
+    options_dataclass = ZorkGrandInquisitorOptions
+    options: ZorkGrandInquisitorOptions
+
+    game = "Zork Grand Inquisitor"
+
+    topology_present = False
+
+    item_name_to_id = item_names_to_id()
+    location_name_to_id = location_names_to_id()
+
+    item_name_groups = item_groups()
+    location_name_groups = location_groups()
+
+    required_client_version: Tuple[int, int, int] = (0, 4, 4)
+
+    web = ZorkGrandInquisitorWebWorld()
+
+    item_name_to_item: Dict[str, ZorkGrandInquisitorItems] = item_names_to_item()
+
+    def create_regions(self) -> None:
+        deathsanity: bool = self.options.deathsanity.value == 1
```

Comment:
as a rule of thumb i really don't like comparing options to arbitrary integers. some alternatives include:
```python
deathsanity: bool = bool(self.options.deathsanity)
```
```python
deathsanity: bool = self.options.deathsanity == DeathSanity.option_true
```
```python
deathsanity: bool = self.options.deathsanity == True
```
```python
deathsanity: bool = self.options.deathsanity == "true"
```

Reply:
I also didn't like that bit but wasn't sure what was possible with the new options API. I'll just use True / False then